packages/flutter/test/widgets/system_context_menu_test.dart

```diff
@@ -1337,4 +1337,80 @@ void main() {
     variant: TargetPlatformVariant.only(TargetPlatform.iOS),
   );
+
+  testWidgets(
+    'Default iOS SystemContextMenu includes Share for non-empty selection',
+    (WidgetTester tester) async {
+      final List<List<IOSSystemContextMenuItemData>> itemsReceived =
+          <List<IOSSystemContextMenuItemData>>[];
+      TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger.setMockMethodCallHandler(
+        SystemChannels.platform,
+        (MethodCall methodCall) async {
+          if (methodCall.method == 'ContextMenu.showSystemContextMenu') {
+            final Map<String, dynamic> arguments = methodCall.arguments as Map<String, dynamic>;
+            final List<dynamic> untypedItems = arguments['items'] as List<dynamic>;
+            final List<IOSSystemContextMenuItemData> lastItems = <IOSSystemContextMenuItemData>[
+              for (final dynamic value in untypedItems)
+                systemContextMenuItemDataFromJson(value as Map<String, dynamic>),
+            ];
+            itemsReceived.add(lastItems);
+          }
+          return;
+        },
+      );
+      addTearDown(() {
+        TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger.setMockMethodCallHandler(
+          SystemChannels.platform,
+          null,
+        );
+      });
+
+      final TextEditingController controller = TextEditingController(text: 'Hello world');
+      addTearDown(controller.dispose);
+      await tester.pumpWidget(
+        MediaQuery(
+          data: const MediaQueryData(supportsShowingSystemContextMenu: true),
+          child: MaterialApp(
+            home: Scaffold(
+              body: Center(
+                child: TextField(
+                  controller: controller,
+                  contextMenuBuilder: (BuildContext context, EditableTextState editableTextState) {
+                    return SystemContextMenu.editableText(editableTextState: editableTextState);
+                  },
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+
+      // Focus the field first (this establishes the TextInputConnection).
+      await tester.tap(find.byType(TextField));
+      await tester.pump();
+      final EditableTextState state = tester.state<EditableTextState>(find.byType(EditableText));
+
+      // Set a non-empty selection to enable sharing.
+      controller.selection = const TextSelection(baseOffset: 0, extentOffset: 5); // "Hello"
+      await tester.pump();
+
+      // Nit: ensure no platform message sent before showing toolbar.
+      expect(itemsReceived, isEmpty);
+
+      // Show the context menu.
+      expect(state.showToolbar(), true);
+      await tester.pump();
+
+      // Assert that the platform message included a Share item.
+      expect(itemsReceived, isNotEmpty);
+      expect(
+        itemsReceived.last.any(
+          (IOSSystemContextMenuItemData e) => e is IOSSystemContextMenuItemDataShare,
+        ),
+        isTrue,
+      );
```

Comment:
Nit: I wonder if there is a matcher that you could use here like `contains`? See https://main-api.flutter.dev/flutter/package-matcher_matcher/.